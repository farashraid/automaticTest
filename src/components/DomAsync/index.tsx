

// import React, { FC, useEffect, useMemo, useState } from "react";
// import { Table } from "antd";

// interface IProps { }

// export const DomAsync: FC<IProps> = ({ }) => {
//     const [text, setText] = useState("");
//     const [tableData, setTableData] = useState<string[]>([]);

//     const hasDescription = useMemo(() => {
//         return text !== "a demo for async test";
//     }, [text]);

//     const fetchData = () => {
//         return new Promise((resolve) => {
//             setTimeout(() => {
//                 resolve("Data from the server");
//             }, 1000); // Simulating a 1-second delay
//         });
//     };


//     const columns = [
//         {
//             title: "Data",
//             dataIndex: "data",
//             key: "data",
//             render: (text: string) => <div>{text}</div>,
//         },
//     ];
//     useEffect(() => {
//         const fetchDataFromServer = async () => {
//             try {
//                 const data = await fetchData();
//                 setText(data as string); // 或者 setText(String(data));
//                 setTableData([data as string]);
//             } catch (error) {
//                 console.error("Error fetching data:", error);
//             }
//         };

//         fetchDataFromServer();
//     }, []);

//     return (
//         <div>
//             <div>{text}</div>
//             {hasDescription && <div>加载中...</div>}
//             {/* <Table dataSource={tableData as string[]}  columns={columns} /> */}
//         </div>
//     );
// };



import { FC, useEffect, useMemo, useState } from "react";
import { Table } from "antd";

interface IProps { }

export const DomAsync: FC<IProps> = ({ }) => {
    const [text, setText] = useState("");
    const columns = [
        {
            title: "Data",
            dataIndex: "data",
            key: "data",
            render: (text: string) => <div>{text}</div>,
        },
    ];
    const hasDescription = useMemo(() => {
        return text !== "a demo for async test";
    }, [text]);
    const [dataSource, setDataSource] = useState([
        { key: '1', name: 'John Doe1', age: 30 },
        { key: '2', name: 'John Doe2', age: 30 },
    ]);
    useEffect(() => {
        const fetchDataFromServer = async () => {
            try {
                const [dataSource, setDataSource] = useState([
                    { key: '1', name: 'John Doe1', age: 30 },
                    { key: '2', name: 'John Doe2', age: 30 },
                    { key: '3', name: 'John Doe3', age: 30 },
                ]);
                // const data = await fetchData();
                // setText(data as string); // 或者 setText(String(data));
                // setTableData([data as string]);
            } catch (error) {
                console.error("Error fetching data:", error);
            }
        };

        fetchDataFromServer();
    }, []);
    return (
        <div>
            <div>{text}</div>
            {hasDescription && <div>加载中...</div>}
            {dataSource.length > 0 && <Table data-testid dataSource={dataSource} columns={columns} />}
        </div>
    );
};