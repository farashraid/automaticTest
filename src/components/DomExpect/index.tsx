import { FC, useState } from "react";
import { Form } from "@douyinfe/semi-ui";

interface IProps {}

// 《6 | DOM断言：页面元素的断言》
export const DomExpect: any = ({}) => {
  const [semiFormValues, setSemiFormValues] = useState({
    username: "zhenmin",
    age: 23,
    sex: "man",
    hobby: "code",
  });
  return (
    <div>
      <div aria-label="empty_note"></div>
      <div role="note" style={{ display: "none" }} aria-hidden>
        1234
      </div>
      <div role="note">1234</div>
      {/* code validation */}
      {/* <div
        role="note"
        style={{ display: "none" }}
        className="test hidden"
        aria-hidden
      >
        1234
      </div> */}
      {/* 虽然它包含 form 角色，但是必须要加上 aria-label 才可以使用 */}
      {/* screen.getByRole("form")进行筛选，这是一个很特殊的规则，因为 form */}
      {/* 元素没有一个可访问的信息 */}
      <form aria-label="form">
        <input
          type="text"
          name="username"
          disabled
          aria-disabled
          defaultValue="zhenmin"
          aria-label="form_username"
        />
        <input
          type="number"
          name="age"
          defaultValue={23}
          required
          aria-label="form_age"
        />
        <input
          type="radio"
          name="sex"
          value="man"
          defaultChecked
          aria-checked
          aria-label="form_sex"
        />
        <input type="radio" name="sex" value="woman" aria-label="form_sex" />
      </form>
      {/* semi 表单验证 */}
      <Form
        initValues={semiFormValues}
        aria-label="semi-form"
        onChange={(data: any): void => {
          setSemiFormValues(data);
        }}
      >
        <Form.Input field="username" disabled name="username" />
        <Form.InputNumber field="age" required name="age" />
        <Form.RadioGroup field="sex" name="sex">
          <Form.Radio value="man" />
          <Form.Radio value="woman" />
        </Form.RadioGroup>
        <Form.Select field="hobby" name="hobby">
          <Form.Select.Option value="code">code</Form.Select.Option>
          <Form.Select.Option value="read">read</Form.Select.Option>
        </Form.Select>
      </Form>
      <input type="hidden" role="note" value={JSON.stringify(semiFormValues)} />
    </div>
  );
};
