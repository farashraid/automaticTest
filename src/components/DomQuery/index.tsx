// ./src/components/DomQuery/index.tsx
import { FC } from "react";

interface IProps {}

// 《4 | DOM查询(上)：页面元素的渲染和行为查询》 & 《5 |DOM查询(下)：页面元素的参照物查询和优先级》
export const DomQuery: FC<IProps> = ({}) => {
  return (
    <div>
      <div>test1</div>
      {/* ... 我们不应该为了选取方便就错误或者重复添加 role，除非是一个自定义的 UI 组件，不然大部分我们需要定位的场景其实都是有默认的 ARIA 语义的。 */}

      <button role="tab">按钮</button>
      <button aria-pressed></button>
      <button aria-describedby="description">
        <div id="description">自定义aria按钮</div>
      </button>
      <div aria-label="test_note">1234</div>
      <label>
        testLabel
        <input />
      </label>
      <input placeholder="a query by placeholder" />
      <input defaultValue="a query by value" readOnly />
      <img alt="a query by alt" />
      <span title="a query by title" />
      <div data-testid="a not so good query"></div>
    </div>
  );
};
