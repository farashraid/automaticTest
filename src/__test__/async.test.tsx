import React from "react";
import {
  render,
  waitFor,
  screen
} from "@testing-library/react";
import type { IContainer } from '../interface';
import { getProvider } from '../provider';
import { DomAsync } from "../components/DomAsync";
import * as table from './table';

Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  });
// 8 | Async 异步：异步方法如何进行单测？
describe("examples for async", () => {
  // ... other content
  
  test("for react testing library", async () => {
    const { container } = render(<DomAsync />);
    // const testDom = await screen.findByText("a demo for async test");
    // const testDom1 = await screen.findByText("John Doe1");
    // expect(testDom).toBeInTheDocument();
    expect(table.query(container)).not.toBeNull();
    
  });
  test('queryRow', () => {
    const { container } = render(<DomAsync />);
    expect(table.queryRow(container,1)).not.toBeNull();
});
});


test('renders text and table with loading message', async () => {
//   const { getByText, queryByText } = render(<DomAsync />);
const tree = render(<DomAsync />);
//   expect(tree).toMatchSnapshot();
  // Check if the initial text is rendered
//   expect(getByText('John Doe1')).toBeInTheDocument();

//   // Check if the loading message is initially displayed
//   expect(getByText('加载中...')).toBeInTheDocument();

//   // Wait for the asynchronous data fetching to complete
  await waitFor(() => {
//     // Check if the loading message is no longer displayed
//     expect(queryByText('加载中...')).toBeNull();
// expect(tree).toMatchSnapshot();
//     // Check if the fetched data is rendered
//     // expect(getByText('John Doe2')).toBeInTheDocument();
  });
});