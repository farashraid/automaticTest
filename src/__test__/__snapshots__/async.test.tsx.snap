// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders text and table with loading message 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div />
        <div>
          加载中...
        </div>
        <div
          class="ant-table-wrapper css-dev-only-do-not-override-2q8sxy"
        >
          <div
            class="ant-spin-nested-loading css-dev-only-do-not-override-2q8sxy"
          >
            <div
              class="ant-spin-container"
            >
              <div
                class="ant-table"
              >
                <div
                  class="ant-table-container"
                >
                  <div
                    class="ant-table-content"
                  >
                    <table
                      style="table-layout: auto;"
                    >
                      <colgroup />
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            Data
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="1"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            <div />
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
              <ul
                class="ant-pagination ant-table-pagination ant-table-pagination-right css-dev-only-do-not-override-2q8sxy"
              >
                <li
                  aria-disabled="true"
                  class="ant-pagination-prev ant-pagination-disabled"
                  title="Previous Page"
                >
                  <button
                    class="ant-pagination-item-link"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      aria-label="left"
                      class="anticon anticon-left"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="left"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                        />
                      </svg>
                    </span>
                  </button>
                </li>
                <li
                  class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                  tabindex="0"
                  title="1"
                >
                  <a
                    rel="nofollow"
                  >
                    1
                  </a>
                </li>
                <li
                  aria-disabled="true"
                  class="ant-pagination-next ant-pagination-disabled"
                  title="Next Page"
                >
                  <button
                    class="ant-pagination-item-link"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      aria-label="right"
                      class="anticon anticon-right"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="right"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                        />
                      </svg>
                    </span>
                  </button>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div />
      <div>
        加载中...
      </div>
      <div
        class="ant-table-wrapper css-dev-only-do-not-override-2q8sxy"
      >
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-2q8sxy"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup />
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          Data
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="1"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div />
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
            <ul
              class="ant-pagination ant-table-pagination ant-table-pagination-right css-dev-only-do-not-override-2q8sxy"
            >
              <li
                aria-disabled="true"
                class="ant-pagination-prev ant-pagination-disabled"
                title="Previous Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="left"
                    class="anticon anticon-left"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="left"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
              <li
                class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                tabindex="0"
                title="1"
              >
                <a
                  rel="nofollow"
                >
                  1
                </a>
              </li>
              <li
                aria-disabled="true"
                class="ant-pagination-next ant-pagination-disabled"
                title="Next Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="right"
                    class="anticon anticon-right"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="right"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders text and table with loading message 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div />
        <div>
          加载中...
        </div>
        <div
          class="ant-table-wrapper css-dev-only-do-not-override-2q8sxy"
        >
          <div
            class="ant-spin-nested-loading css-dev-only-do-not-override-2q8sxy"
          >
            <div
              class="ant-spin-container"
            >
              <div
                class="ant-table"
              >
                <div
                  class="ant-table-container"
                >
                  <div
                    class="ant-table-content"
                  >
                    <table
                      style="table-layout: auto;"
                    >
                      <colgroup />
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            Data
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="1"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            <div />
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
              <ul
                class="ant-pagination ant-table-pagination ant-table-pagination-right css-dev-only-do-not-override-2q8sxy"
              >
                <li
                  aria-disabled="true"
                  class="ant-pagination-prev ant-pagination-disabled"
                  title="Previous Page"
                >
                  <button
                    class="ant-pagination-item-link"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      aria-label="left"
                      class="anticon anticon-left"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="left"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                        />
                      </svg>
                    </span>
                  </button>
                </li>
                <li
                  class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                  tabindex="0"
                  title="1"
                >
                  <a
                    rel="nofollow"
                  >
                    1
                  </a>
                </li>
                <li
                  aria-disabled="true"
                  class="ant-pagination-next ant-pagination-disabled"
                  title="Next Page"
                >
                  <button
                    class="ant-pagination-item-link"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      aria-label="right"
                      class="anticon anticon-right"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="right"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                        />
                      </svg>
                    </span>
                  </button>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div />
      <div>
        加载中...
      </div>
      <div
        class="ant-table-wrapper css-dev-only-do-not-override-2q8sxy"
      >
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-2q8sxy"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup />
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          Data
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="1"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div />
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
            <ul
              class="ant-pagination ant-table-pagination ant-table-pagination-right css-dev-only-do-not-override-2q8sxy"
            >
              <li
                aria-disabled="true"
                class="ant-pagination-prev ant-pagination-disabled"
                title="Previous Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="left"
                    class="anticon anticon-left"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="left"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
              <li
                class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                tabindex="0"
                title="1"
              >
                <a
                  rel="nofollow"
                >
                  1
                </a>
              </li>
              <li
                aria-disabled="true"
                class="ant-pagination-next ant-pagination-disabled"
                title="Next Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="right"
                    class="anticon anticon-right"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="right"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
